### PR TITLE
Add deep merge package

### PR DESCRIPTION
**Pull request by pmareke**
Move `deep-merge` package to the mono repository

### CODE REVIEW

File: packages/eb-deep-merge/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1 @@
+npmPackagePipeline {}
```
Comment (LuisMartinez15) [inline packages/eb-deep-merge/Jenkinsfile:1]: Is this file still necessary? I don't think so, can we get rid of it?

File: packages/eb-deep-merge/tslint.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "rulesDirectory": [
+    "node_modules/tslint"
+  ],
+  "extends": [
+    "@empathybroker/eb-tslint"
+  ]
+}
```
Comment (LuisMartinez15) [inline packages/eb-deep-merge/tslint.json:8]: Remove unnecessary file.

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]

File: packages/eb-deep-merge/tslint.json (hunk 2 above)
Comment (LuisMartinez15) [reply packages/eb-deep-merge/tslint.json:8]: Can we rename the folder from `eb-deep-merge` to just `deep-merge`?

File: packages/eb-deep-merge/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,4878 @@
+{
+  "name": "@empathybroker/deep-merge",
```
Comment (javieri-empathy) [inline packages/eb-deep-merge/package-lock.json:2]: Do a `npm install` from the root of the monorepo so this package-lock is updated.

File: packages/eb-deep-merge/Jenkinsfile (hunk 1 above)
Comment (pmareke) [reply packages/eb-deep-merge/Jenkinsfile:1]: Fix in the other PR.

File: packages/eb-deep-merge/tslint.json (hunk 2 above)
Comment (pmareke) [reply packages/eb-deep-merge/tslint.json:8]: Fix in the other PR.

File: packages/eb-deep-merge/package-lock.json (hunk 3 above)
Comment (pmareke) [reply packages/eb-deep-merge/package-lock.json:2]: Sorry my fault.

**Review by javieri-empathy — APPROVED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Review by LuisMartinez15 — APPROVED** [review]
